src/arith/solve_linear_inequality.cc

```diff
@@ -0,0 +1,648 @@
+    equal.reserve(std::min(upper_bounds.size(), lower_bounds.size()));
+    MoveEquality(&upper_bounds, &lower_bounds, &equal);
+    std::vector<PrimExpr> equal_list(equal.begin(), equal.end());
+    std::sort(equal_list.begin(), equal_list.end(), ExprLess());
+
+    // Write it to the result.
+    IntGrpBounds bnds(make_const(v.dtype(), coef_lcm),
+                      Array<PrimExpr>(lower_bounds.begin(), lower_bounds.end()),
+                      Array<PrimExpr>(equal_list.begin(), equal_list.end()),
+                      Array<PrimExpr>(upper_bounds.begin(), upper_bounds.end()));
+    res_bounds.Set(v, bnds);
+
+    std::swap(current_ineq_set_to_solve, next_ineq_set_to_solve);
+  }
+
+  // Everything that is left goes to res.relations
+  Array<PrimExpr> other_conditions;
+  for (const PrimExpr& e : current_ineq_set_to_solve) {
+    PrimExpr e_simp = analyzer.Simplify(e, 3);
+    if (is_const_int(e_simp, 0)) {
+      // contradiction detected
+      other_conditions = {const_false()};
+      break;
+    } else if (is_const_int(e_simp, 1)) {
+      continue;
+    } else {
+      other_conditions.push_back(e_simp);
+    }
+  }
+
+  for (const PrimExpr& e : rest) {
+    other_conditions.push_back(e);
+  }
+
+  return {res_bounds, other_conditions};
+}
+
+IntConstraints SolveInequalitiesToRange(const IntConstraints& inequalities) {
+  // Resulting ranges will contain ranges for the new variables and for the variables that are
+  // not in the inequalities->variables but are in inequalities->ranges
+  // It will be useful when solving Jacobian axes jac_xxx)
+  Map<Var, Range> res_ranges;
+  // we get a set of equality, lower, upper bound of each variable.
+  auto solved_system = SolveLinearInequalities(inequalities);
+
+  Map<Var, IntGrpBounds> solved_bounds = solved_system.first;
+  Array<PrimExpr> solved_other_relations = solved_system.second;
+
+  Array<PrimExpr> res_relations;
+
+  // this keeps being updated during determining the range of each variable.
+  Map<Var, Range> vranges;
+  for (std::pair<Var, Range> vr : inequalities->ranges) {
+    vranges.Set(vr.first, vr.second);
+  }
+
+  // We process variables in the reverse direction to start with the most independent one.
+  // This order is needed to compute new ranges.
+  for (auto it = inequalities->variables.rbegin(); it != inequalities->variables.rend(); ++it) {
+    arith::Analyzer analyzer;
+    analyzer.Bind(vranges);
+
+    const Var& var = *it;
+    CHECK(solved_bounds.count(var));
+    auto bnd = solved_bounds[var];
+    if (is_one(bnd->coef) && !bnd->equal.empty()) {
+      // There is an equation of the form `v == expr`, so this variable can be completely removed.
+      // Note that we use the 0-th expression because they are ordered by complexity,
+      // so it must be the simplest one.
+      Range best_range(bnd->equal[0], analyzer.Simplify(bnd->equal[0] + 1, 3));
+      res_ranges.Set(var, best_range);
```

Comment:
nit: res_ranges.Set & vranges.Set can be taken to common place below if...else... block.

Reply:
in the else branch, they are not set unless `best_range` is defined.